clusterloader/cluster_loader.go

```diff
@@ -0,0 +1,189 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterloader
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"time"
+
+	clframe "github.com/kubernetes/perf-tests/clusterloader/framework"
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	"k8s.io/kubernetes/pkg/api/v1"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/release_1_5"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/test/e2e/framework"
+	testutils "k8s.io/kubernetes/test/utils"
+)
+
+var _ = framework.KubeDescribe("Cluster Loader [Feature:ManualPerformance]", func() {
+	f := framework.NewDefaultFramework("cluster-loader")
+	defer ginkgo.GinkgoRecover()
+
+	var c clientset.Interface
+	ginkgo.BeforeEach(func() {
+		c = f.ClientSet
+	})
+
+	ginkgo.It(fmt.Sprintf("running config file"), func() {
+		project := clframe.ConfigContext.ClusterLoader.Projects
+		tuningSets := clframe.ConfigContext.ClusterLoader.TuningSets
+		if len(project) < 1 {
+			framework.Failf("invalid config file.\nFile: %v", project)
+		}
+
+		var namespaces []*v1.Namespace
+		//totalPods := 0 // Keep track of how many pods for stepping
+		// TODO sjug: add concurrency
+		for _, p := range project {
+			// Find tuning if we have it
+			tuning := getTuningSet(tuningSets, p.Tuning)
+			framework.Logf("Our tuning set is: %v", tuning)
+			for j := 0; j < p.Number; j++ {
+				// Create namespaces as defined in Cluster Loader config
+				nsName := appendIntToString(p.Basename, j)
+				ns, err := f.CreateNamespace(nsName, nil)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				framework.Logf("%d/%d : Created new namespace: %v", j+1, p.Number, nsName)
+				namespaces = append(namespaces, ns)
+
+				// Create templates as defined
+				for _, v := range p.Templates {
+					createTemplate(v.Basename, ns, mkPath(v.File), v.Number, tuning)
+				}
+				// This is too familiar, create pods
+				for _, v := range p.Pods {
+					config := parsePods(mkPath(v.File))
+					clframe.CreatePods(f, v.Basename, ns.Name, config.Spec, v.Number, tuning)
```

Comment:
That's exactly what I do not want. If the pod fails I don't want it being restarted because it should not fail. When the test completes the pod should stop and this signals the end of the test to the tool. Is there a way to use the RC/RS/Dep to start the pod and then go away and not restart it?